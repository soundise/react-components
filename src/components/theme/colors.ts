export const colors = {
  primary: {
    main: '#FFFFFF',
  },
  secondary: {
    main: '#000000',
  },
  tertiary: {
    main: '#8D71FF',
  },
};
