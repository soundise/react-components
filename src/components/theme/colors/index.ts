export * from './neutral';
export * from './primary';
export * from './gray';
export * from './red';
export * from './green';
