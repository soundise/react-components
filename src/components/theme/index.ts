export type { ITheme } from './theme';
export { darkTheme } from './darkTheme';
export { lightTheme } from './lightTheme';
export { GlobalStyle } from './GlobalStyle';
