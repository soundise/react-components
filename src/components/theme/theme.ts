export interface ITheme {
  colors: {
    background: string;
    foreground: string;
    primary: {
      main: string;
    };
    secondary: {
      main: string;
    };
    tertiary: {
      main: string;
    };
  };
}
