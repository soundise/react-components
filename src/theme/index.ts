export type { ITheme } from './theme';
export { darkTheme } from './darkTheme';
export { lightTheme } from './lightTheme';
export * from './typography';
export * from './colors';
export * from './breakpoints';
