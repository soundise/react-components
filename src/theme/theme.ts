export interface ITheme {
  name: string;
  variant: 'light' | 'dark';
  typography: {
    fonts: {
      primary: string;
      fallback: string;
      code: {
        primary: string;
        fallback: string;
      };
    };
    weights: {
      regular: number;
      bold: number;
    };
  };
  breakpoints: {
    mobileLandscape: string;
    tablet: string;
    desktopSmall: string;
    desktopLarge: string;
  };
  colors: {
    background: string;
    foreground: string;
    border: string;
    text: {
      primary: string;
      secondary: string;
      tertiary: string;
    };
    status: {
      success: string;
      warning: string;
      error: string;
    };
    brand: {
      primary: string;
      secondary: string;
      tertiary: string;
    };
    neutral: {
      white: string;
      black: string;
    };
    primary: {
      pr050: string;
      pr100: string;
      pr200: string;
      pr300: string;
      pr400: string;
      pr500: string;
      pr600: string;
      pr700: string;
      pr800: string;
      pr900: string;
    };
    gray: {
      gr050: string;
      gr100: string;
      gr200: string;
      gr300: string;
      gr400: string;
      gr500: string;
      gr600: string;
      gr700: string;
      gr800: string;
      gr900: string;
    };
    green: {
      g050: string;
      g100: string;
      g200: string;
      g300: string;
      g400: string;
      g500: string;
      g600: string;
      g700: string;
      g800: string;
      g900: string;
    };
    yellow: {
      y050: string;
      y100: string;
      y200: string;
      y300: string;
      y400: string;
      y500: string;
      y600: string;
      y700: string;
      y800: string;
      y900: string;
    };
    red: {
      r050: string;
      r100: string;
      r200: string;
      r300: string;
      r400: string;
      r500: string;
      r600: string;
      r700: string;
      r800: string;
      r900: string;
    };
  };
}
