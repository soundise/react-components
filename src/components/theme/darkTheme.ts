import type { ITheme } from './theme';
import * as typography from './typography';
import * as colors from './colors';
import { breakpoints } from './breakpoints';

export const darkTheme: ITheme = {
  typography: {
    ...typography,
  },
  colors: {
    background: colors.neutral.black,
    foreground: colors.neutral.white,
    textPrimary: colors.neutral.white,
    textSecondary: colors.neutral.black,
    textTertiary: colors.primary.pr500,
    ...colors,
  },
  breakpoints: {
    ...breakpoints,
  },
};
