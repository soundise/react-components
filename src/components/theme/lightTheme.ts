import type { ITheme } from './theme';
import * as colors from './colors';

export const lightTheme: ITheme = {
  colors: {
    background: colors.neutral.white,
    foreground: colors.neutral.black,
    ...colors,
  },
};
