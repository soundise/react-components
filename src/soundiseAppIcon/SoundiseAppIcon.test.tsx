import React from 'react';
import { render, cleanup } from 'tests';
import 'jest-styled-components';
import { SoundiseAppIcon } from './SoundiseAppIcon';
import { darkTheme } from '../theme';

afterEach(cleanup);

describe('SoundiseAppIcon', () => {
  it('should render the Soundise app icon as an SVG', () => {
    const { getByTestId } = render(<SoundiseAppIcon />);
    const soundiseAppIcon = getByTestId('soundise-app-icon');

    expect(soundiseAppIcon).toBeInTheDocument();
    expect(soundiseAppIcon.tagName).toBe('svg');
    expect(soundiseAppIcon).toMatchSnapshot();
  });

  it('should render primary Soundise app icon', () => {
    const { getAllByTestId, getByTestId } = render(
      <SoundiseAppIcon variant="primary" />
    );
    const soundiseAppIconPaths = getAllByTestId('soundise-app-icon-path');
    const soundiseAppIconRect = getByTestId('soundise-app-icon-rect');

    soundiseAppIconPaths.forEach((path) => {
      expect(path).toHaveStyleRule('fill', darkTheme.colors.textSecondary);
    });
    expect(soundiseAppIconRect).toHaveStyleRule(
      'fill',
      darkTheme.colors.textPrimary
    );
    expect(soundiseAppIconPaths).toMatchSnapshot();
    expect(soundiseAppIconRect).toMatchSnapshot();
  });

  it('should render secondary Soundise app icon', () => {
    const { getAllByTestId, getByTestId } = render(
      <SoundiseAppIcon variant="secondary" />
    );
    const soundiseAppIconPaths = getAllByTestId('soundise-app-icon-path');
    const soundiseAppIconRect = getByTestId('soundise-app-icon-rect');

    soundiseAppIconPaths.forEach((path) => {
      expect(path).toHaveStyleRule('fill', darkTheme.colors.textPrimary);
    });
    expect(soundiseAppIconRect).toHaveStyleRule(
      'fill',
      darkTheme.colors.textSecondary
    );
    expect(soundiseAppIconPaths).toMatchSnapshot();
    expect(soundiseAppIconRect).toMatchSnapshot();
  });
});
