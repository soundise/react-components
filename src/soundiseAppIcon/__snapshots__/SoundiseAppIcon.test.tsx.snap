// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundiseAppIcon should render primary Soundise app icon 1`] = `
[
  .c0 {
  fill: #000000;
}

<path
    class="c0"
    d="M128.833 142.187C127.741 142.187 126.66 142.404 125.651 142.826C124.643 143.248 123.726 143.867 122.954 144.646C122.182 145.426 121.569 146.351 121.151 147.37C120.733 148.388 120.518 149.48 120.518 150.582V169.348C120.522 170.168 120.356 170.981 120.029 171.732C119.702 172.483 119.222 173.156 118.621 173.708C118.019 174.26 117.31 174.678 116.538 174.936C115.766 175.193 114.95 175.283 114.141 175.201C112.681 175.032 111.334 174.322 110.363 173.208C109.392 172.094 108.865 170.656 108.884 169.172V126.134C108.882 123.123 108.285 120.142 107.129 117.366C105.973 114.59 104.281 112.075 102.152 109.966C100.023 107.857 97.4994 106.198 94.7283 105.084C91.9572 103.971 88.9941 103.426 86.0121 103.481C73.7257 103.715 64 114.105 64 126.539V138.865C64.0009 141.063 64.8557 143.173 66.3808 144.74C67.9058 146.308 69.9793 147.21 72.1555 147.251C76.813 147.338 80.6299 143.348 80.6299 138.641V126.134C80.6258 125.313 80.7926 124.501 81.1196 123.75C81.4465 122.999 81.9262 122.325 82.5276 121.773C83.1289 121.221 83.8385 120.803 84.6103 120.546C85.382 120.289 86.1988 120.198 87.0074 120.28C88.466 120.449 89.8107 121.158 90.7817 122.27C91.7526 123.382 92.2808 124.818 92.264 126.3V169.343C92.2674 172.354 92.8645 175.334 94.0206 178.11C95.1766 180.885 96.8684 183.401 98.9974 185.509C101.126 187.618 103.65 189.278 106.421 190.391C109.192 191.505 112.154 192.05 115.136 191.996C127.423 191.762 137.148 181.367 137.148 168.962V150.582C137.148 148.356 136.272 146.221 134.713 144.646C133.154 143.072 131.039 142.187 128.833 142.187V142.187Z"
    data-testid="soundise-app-icon-path"
    variant="primary"
  />,
  .c0 {
  fill: #000000;
}

<path
    class="c0"
    d="M191.985 111.081C191.876 108.932 190.953 106.908 189.408 105.427C187.862 103.946 185.811 103.122 183.68 103.125C179.11 103.125 175.37 107.11 175.37 111.73V151.29C175.369 152.05 175.21 152.802 174.905 153.497C174.599 154.191 174.154 154.814 173.596 155.325C173.039 155.836 172.381 156.224 171.667 156.464C170.953 156.704 170.196 156.791 169.447 156.719C168.092 156.561 166.843 155.902 165.942 154.868C165.041 153.835 164.551 152.501 164.567 151.124V106.857C164.567 106.808 164.567 106.759 164.567 106.71V86.6131C164.567 74.4183 155.034 64.2429 142.98 64.0039C140.052 63.9488 137.143 64.483 134.421 65.5752C131.7 66.6675 129.221 68.2959 127.13 70.3656C125.039 72.4352 123.377 74.9046 122.241 77.6297C121.105 80.3548 120.518 83.2809 120.514 86.2375V124.232C120.514 126.458 121.39 128.594 122.949 130.168C124.508 131.742 126.623 132.627 128.829 132.627C131.034 132.627 133.149 131.742 134.708 130.168C136.268 128.594 137.144 126.458 137.144 124.232V86.2375C137.144 85.4771 137.302 84.7251 137.607 84.0299C137.913 83.3347 138.358 82.7115 138.916 82.2004C139.474 81.6892 140.131 81.3013 140.846 81.0616C141.561 80.8219 142.317 80.7357 143.067 80.8084C144.421 80.9652 145.67 81.6236 146.571 82.6562C147.473 83.6888 147.963 85.0222 147.947 86.3985V104.681C147.947 104.725 147.947 104.774 147.947 104.823V151.285C147.95 154.24 148.536 157.166 149.671 159.89C150.806 162.615 152.467 165.084 154.557 167.153C156.647 169.223 159.124 170.851 161.844 171.944C164.563 173.037 167.472 173.572 170.398 173.519C182.477 173.289 192 163.09 192 150.914V111.515C191.995 111.369 191.99 111.222 191.985 111.081Z"
    data-testid="soundise-app-icon-path"
    variant="primary"
  />,
]
`;

exports[`SoundiseAppIcon should render primary Soundise app icon 2`] = `
.c0 {
  fill: #FFFFFF;
}

<rect
  class="c0"
  data-testid="soundise-app-icon-rect"
  height="256"
  rx="45"
  variant="primary"
  width="256"
/>
`;

exports[`SoundiseAppIcon should render secondary Soundise app icon 1`] = `
[
  .c0 {
  fill: #000000;
  fill: #FFFFFF;
}

<path
    class="c0"
    d="M128.833 142.187C127.741 142.187 126.66 142.404 125.651 142.826C124.643 143.248 123.726 143.867 122.954 144.646C122.182 145.426 121.569 146.351 121.151 147.37C120.733 148.388 120.518 149.48 120.518 150.582V169.348C120.522 170.168 120.356 170.981 120.029 171.732C119.702 172.483 119.222 173.156 118.621 173.708C118.019 174.26 117.31 174.678 116.538 174.936C115.766 175.193 114.95 175.283 114.141 175.201C112.681 175.032 111.334 174.322 110.363 173.208C109.392 172.094 108.865 170.656 108.884 169.172V126.134C108.882 123.123 108.285 120.142 107.129 117.366C105.973 114.59 104.281 112.075 102.152 109.966C100.023 107.857 97.4994 106.198 94.7283 105.084C91.9572 103.971 88.9941 103.426 86.0121 103.481C73.7257 103.715 64 114.105 64 126.539V138.865C64.0009 141.063 64.8557 143.173 66.3808 144.74C67.9058 146.308 69.9793 147.21 72.1555 147.251C76.813 147.338 80.6299 143.348 80.6299 138.641V126.134C80.6258 125.313 80.7926 124.501 81.1196 123.75C81.4465 122.999 81.9262 122.325 82.5276 121.773C83.1289 121.221 83.8385 120.803 84.6103 120.546C85.382 120.289 86.1988 120.198 87.0074 120.28C88.466 120.449 89.8107 121.158 90.7817 122.27C91.7526 123.382 92.2808 124.818 92.264 126.3V169.343C92.2674 172.354 92.8645 175.334 94.0206 178.11C95.1766 180.885 96.8684 183.401 98.9974 185.509C101.126 187.618 103.65 189.278 106.421 190.391C109.192 191.505 112.154 192.05 115.136 191.996C127.423 191.762 137.148 181.367 137.148 168.962V150.582C137.148 148.356 136.272 146.221 134.713 144.646C133.154 143.072 131.039 142.187 128.833 142.187V142.187Z"
    data-testid="soundise-app-icon-path"
    variant="secondary"
  />,
  .c0 {
  fill: #000000;
  fill: #FFFFFF;
}

<path
    class="c0"
    d="M191.985 111.081C191.876 108.932 190.953 106.908 189.408 105.427C187.862 103.946 185.811 103.122 183.68 103.125C179.11 103.125 175.37 107.11 175.37 111.73V151.29C175.369 152.05 175.21 152.802 174.905 153.497C174.599 154.191 174.154 154.814 173.596 155.325C173.039 155.836 172.381 156.224 171.667 156.464C170.953 156.704 170.196 156.791 169.447 156.719C168.092 156.561 166.843 155.902 165.942 154.868C165.041 153.835 164.551 152.501 164.567 151.124V106.857C164.567 106.808 164.567 106.759 164.567 106.71V86.6131C164.567 74.4183 155.034 64.2429 142.98 64.0039C140.052 63.9488 137.143 64.483 134.421 65.5752C131.7 66.6675 129.221 68.2959 127.13 70.3656C125.039 72.4352 123.377 74.9046 122.241 77.6297C121.105 80.3548 120.518 83.2809 120.514 86.2375V124.232C120.514 126.458 121.39 128.594 122.949 130.168C124.508 131.742 126.623 132.627 128.829 132.627C131.034 132.627 133.149 131.742 134.708 130.168C136.268 128.594 137.144 126.458 137.144 124.232V86.2375C137.144 85.4771 137.302 84.7251 137.607 84.0299C137.913 83.3347 138.358 82.7115 138.916 82.2004C139.474 81.6892 140.131 81.3013 140.846 81.0616C141.561 80.8219 142.317 80.7357 143.067 80.8084C144.421 80.9652 145.67 81.6236 146.571 82.6562C147.473 83.6888 147.963 85.0222 147.947 86.3985V104.681C147.947 104.725 147.947 104.774 147.947 104.823V151.285C147.95 154.24 148.536 157.166 149.671 159.89C150.806 162.615 152.467 165.084 154.557 167.153C156.647 169.223 159.124 170.851 161.844 171.944C164.563 173.037 167.472 173.572 170.398 173.519C182.477 173.289 192 163.09 192 150.914V111.515C191.995 111.369 191.99 111.222 191.985 111.081Z"
    data-testid="soundise-app-icon-path"
    variant="secondary"
  />,
]
`;

exports[`SoundiseAppIcon should render secondary Soundise app icon 2`] = `
.c0 {
  fill: #FFFFFF;
  fill: #000000;
}

<rect
  class="c0"
  data-testid="soundise-app-icon-rect"
  height="256"
  rx="45"
  variant="secondary"
  width="256"
/>
`;

exports[`SoundiseAppIcon should render the Soundise app icon as an SVG 1`] = `
.c0 {
  fill: #FFFFFF;
}

.c1 {
  fill: #000000;
}

<svg
  data-testid="soundise-app-icon"
  fill="none"
  height="64"
  viewBox="0 0 256 256"
  width="64"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    class="c0"
    data-testid="soundise-app-icon-rect"
    height="256"
    rx="45"
    variant="primary"
    width="256"
  />
  <path
    class="c1"
    d="M128.833 142.187C127.741 142.187 126.66 142.404 125.651 142.826C124.643 143.248 123.726 143.867 122.954 144.646C122.182 145.426 121.569 146.351 121.151 147.37C120.733 148.388 120.518 149.48 120.518 150.582V169.348C120.522 170.168 120.356 170.981 120.029 171.732C119.702 172.483 119.222 173.156 118.621 173.708C118.019 174.26 117.31 174.678 116.538 174.936C115.766 175.193 114.95 175.283 114.141 175.201C112.681 175.032 111.334 174.322 110.363 173.208C109.392 172.094 108.865 170.656 108.884 169.172V126.134C108.882 123.123 108.285 120.142 107.129 117.366C105.973 114.59 104.281 112.075 102.152 109.966C100.023 107.857 97.4994 106.198 94.7283 105.084C91.9572 103.971 88.9941 103.426 86.0121 103.481C73.7257 103.715 64 114.105 64 126.539V138.865C64.0009 141.063 64.8557 143.173 66.3808 144.74C67.9058 146.308 69.9793 147.21 72.1555 147.251C76.813 147.338 80.6299 143.348 80.6299 138.641V126.134C80.6258 125.313 80.7926 124.501 81.1196 123.75C81.4465 122.999 81.9262 122.325 82.5276 121.773C83.1289 121.221 83.8385 120.803 84.6103 120.546C85.382 120.289 86.1988 120.198 87.0074 120.28C88.466 120.449 89.8107 121.158 90.7817 122.27C91.7526 123.382 92.2808 124.818 92.264 126.3V169.343C92.2674 172.354 92.8645 175.334 94.0206 178.11C95.1766 180.885 96.8684 183.401 98.9974 185.509C101.126 187.618 103.65 189.278 106.421 190.391C109.192 191.505 112.154 192.05 115.136 191.996C127.423 191.762 137.148 181.367 137.148 168.962V150.582C137.148 148.356 136.272 146.221 134.713 144.646C133.154 143.072 131.039 142.187 128.833 142.187V142.187Z"
    data-testid="soundise-app-icon-path"
    variant="primary"
  />
  <path
    class="c1"
    d="M191.985 111.081C191.876 108.932 190.953 106.908 189.408 105.427C187.862 103.946 185.811 103.122 183.68 103.125C179.11 103.125 175.37 107.11 175.37 111.73V151.29C175.369 152.05 175.21 152.802 174.905 153.497C174.599 154.191 174.154 154.814 173.596 155.325C173.039 155.836 172.381 156.224 171.667 156.464C170.953 156.704 170.196 156.791 169.447 156.719C168.092 156.561 166.843 155.902 165.942 154.868C165.041 153.835 164.551 152.501 164.567 151.124V106.857C164.567 106.808 164.567 106.759 164.567 106.71V86.6131C164.567 74.4183 155.034 64.2429 142.98 64.0039C140.052 63.9488 137.143 64.483 134.421 65.5752C131.7 66.6675 129.221 68.2959 127.13 70.3656C125.039 72.4352 123.377 74.9046 122.241 77.6297C121.105 80.3548 120.518 83.2809 120.514 86.2375V124.232C120.514 126.458 121.39 128.594 122.949 130.168C124.508 131.742 126.623 132.627 128.829 132.627C131.034 132.627 133.149 131.742 134.708 130.168C136.268 128.594 137.144 126.458 137.144 124.232V86.2375C137.144 85.4771 137.302 84.7251 137.607 84.0299C137.913 83.3347 138.358 82.7115 138.916 82.2004C139.474 81.6892 140.131 81.3013 140.846 81.0616C141.561 80.8219 142.317 80.7357 143.067 80.8084C144.421 80.9652 145.67 81.6236 146.571 82.6562C147.473 83.6888 147.963 85.0222 147.947 86.3985V104.681C147.947 104.725 147.947 104.774 147.947 104.823V151.285C147.95 154.24 148.536 157.166 149.671 159.89C150.806 162.615 152.467 165.084 154.557 167.153C156.647 169.223 159.124 170.851 161.844 171.944C164.563 173.037 167.472 173.572 170.398 173.519C182.477 173.289 192 163.09 192 150.914V111.515C191.995 111.369 191.99 111.222 191.985 111.081Z"
    data-testid="soundise-app-icon-path"
    variant="primary"
  />
</svg>
`;
